File: toast/android/src/main/java/com/capacitorjs/plugins/toast/ToastPlugin.java
package com.capacitorjs.plugins.toast;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Toast")
public class ToastPlugin extends Plugin {

    @PluginMethod
    public void show(PluginCall call) {
        String text = call.getString("text");
        if (text == null) {
            call.reject("Must provide text");
            return;
        }

        String durationType = call.getString("duration", "short");

        int duration = android.widget.Toast.LENGTH_SHORT;
        if ("long".equals(durationType)) {
            duration = android.widget.Toast.LENGTH_LONG;
        }
        String position = call.getString("position", "bottom");
        Toast.show(getContext(), text, duration, position);

        call.resolve();
    }
}

File: toast/ios/Plugin/ToastPlugin.swift
import Foundation
import Capacitor

@objc(ToastPlugin)
public class ToastPlugin: CAPPlugin {

    @objc func show(_ call: CAPPluginCall) {
        guard let text = call.getString("text") else {
            call.reject("text must be provided and must be a string.")
            return
        }
        let durationType = call.getString("duration", "short")
        let duration = durationType == "long" ? 3500 : 2000
        let position = call.getString("position", "bottom")

        guard let viewController = bridge?.viewController else {
            call.reject("Unable to display toast!")
            return
        }
        Toast.showToast(in: viewController, text: text, duration: duration, position: position, completion: {(_) in
            call.resolve()
        })
    }
}
